src/v/cloud_storage/partition_manifest.cc

```diff
@@ -928,41 +928,118 @@ partition_manifest partition_manifest::truncate() {
     return removed;
 }
 
-partition_manifest partition_manifest::spillover(model::offset start_offset) {
+void partition_manifest::spillover(const segment_meta& spillover_meta) {
+    auto start_offset = model::next_offset(spillover_meta.committed_offset);
+    auto append_tx = _spillover_manifests.append(spillover_meta);
     if (!advance_start_offset(start_offset)) {
         throw std::runtime_error(fmt_with_ctx(
           fmt::format,
           "{} can't apply spillover to manifest, offset: {}",
           _ntp,
           start_offset));
     }
+
     auto removed = truncate();
```

Comment:
It's validated outside. The caller is supposed to invoke the method that checks the metadata. The validation downstream only results in error logging. It's tricky to do the other way because `truncate` returns the manifest and we're generating metadata from it. Because of that we can't do this before it's truncated.

Reply:
I guess this is fine. I'm just a bit weary of the fact that we can roll back the addition of the spillover manifest and still truncate the manifest. A way around this would be to make `truncate()` a two step process.

Reply:
I think I fixed this now